frontend/packages/console-shared/src/utils/resource-utils.ts

```diff
@@ -410,6 +416,21 @@ const getRolloutStatus = (
   return notFailedOrCancelled && previous && previous.pods.length > 0;
 };
 
+const deploymentIsInProgress = (resource: K8sResourceKind): boolean => {
+  return (
+    [
+      DEPLOYMENT_PHASE.new,
+      DEPLOYMENT_PHASE.pending,
+      DEPLOYMENT_PHASE.running,
+      DEPLOYMENT_PHASE.complete,
+    ].indexOf(getDeploymentPhase(resource)) > -1
+  );
+};
+
+const isReplicationControllerVisible = (resource: K8sResourceKind): boolean => {
+  return _.get(resource, ['status', 'replicas'], false) || deploymentIsInProgress(resource);
```

Comment:
we could simply do
```suggestion
  return _.get(resource, ['status', 'replicas'], deploymentIsInProgress(resource));
```

Reply:
The return type is lying here since it might return a number instead of a boolean.

```suggestion
  return !!_.get(resource, ['status', 'replicas']) || deploymentIsInProgress(resource);
```

Reply:
Won't this return true for all previous "complete" deployments?

Reply:
fixed in https://github.com/openshift/console/pull/3627/commits/e0d1c431754ec9c3ea4b8b29025478d761bc501f